code/modules/power/supermatter/supermatter.dm

```diff
@@ -510,7 +510,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			air_update_turf()
 
 	for(var/mob/living/carbon/human/l in viewers(HALLUCINATION_RANGE(power), src)) // If they can see it without mesons on.  Bad on them.
-		if(HAS_TRAIT(l, TRAIT_MADNESS_IMMUNE) || HAS_TRAIT(l.mind, TRAIT_MADNESS_IMMUNE))
+		if(HAS_TRAIT(l, TRAIT_MADNESS_IMMUNE) || l.mind && HAS_TRAIT(l.mind, TRAIT_MADNESS_IMMUNE))
```

Comment:
```suggestion
		if(HAS_TRAIT(l, TRAIT_MADNESS_IMMUNE) || (l.mind && HAS_TRAIT(l.mind, TRAIT_MADNESS_IMMUNE)))
```
Can we bracket this expression, just to be 100% sure and clear where the precedence is going to wrap things?

Reply:
Sure. I wasn't aware that this was the appropriate way to do it.